src/main/resources/META-INF/reaktivity/mqtt.idl

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2016-2019 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+scope mqtt
+{
+    struct MqttSubscribeInfo
+    {
+        uint16 packetId;
+    }
+
+    struct MqttTopicFormat
+    {
+        string topic;
+        uint8 options;
+    }
+
+    struct MqttUnsubscribePacket
+    {
+        list<string> topics;
+        list<mqtt::MqttUserProperty> userProperties;
+    }
+
+    struct MqttPublishPacket
+    {
+        uint16 packetId;
+        string topic;
+        uint32 messageExpiry;
+        mqtt::MqttResponseTopicFormat respTopicInfo;
+        list<mqtt::MqttUserProperty> properties;
+        string contentType;
+        mqtt::MqttPayloadFormat payloadFormat;
+    }
+
+    struct MqttResponseTopicFormat
+    {
+        string topic;
+        mqtt::MqttBinary correlationInfo;
```

Comment:
Is correlation info needed on the topic?
Normally correlation is used on a per-message basis to implement request-response over messaging where the responses may not arrive back in the same order as the request messages were sent, so need to be correlated to their corresponding request by correlation info.